### DISCUSSION TITLE
How do you know/measure your throughput?

### DISCUSSION THREAD

**Discussion by jeewee00** [jeroenterheerdt/HAsmartirrigation#485] [General]
Noob-alert: I have a rainbird ESP-TM2 system with 3 zones (2 for lawns, 1 for hedges and borders) and a Home Assistant. I'd really like to get the HA integration running and irrigate my garden zones with the right amounts. But I have no clue where to find or measure the throughput. It's not in the Rainbird app and I assume it all depends on the pump type, setting, number of nozzles, etc. So is there a way to measure it myself?

**Comment by jeroenterheerdt** [comment]
you can either consult your documentation - my sprinkler system has indicators on each sprinkler head that tell me the throughput in lpm for each sprinkler head, so from there on it's easy math. Or search how to measure sprinkler system throughput to see advice involving running the sprinklers for a set amount of time and measuring the amount of water that they put out.

**Comment by lynxchat4** [comment]
If you use water from your own tanks, then it is enough to measure how long it takes the system to use 10 - 50 - 100 liters of measured water. But it is best to install a water meter with a pulse output. Or a simple one and measure manually.

**Comment by lhsans** [comment]
I have a similar setup (rainbird nozzles with ESP-TM2 controller). From what I can see in the rainbird nozzle charts, you have either a  l/m value or the mm/h value based on your nozzle configuration (50% overlap, square or triangular positioning).

I calculated both for multiple zones:

- l/m value: sum all l/m values (based on my crude radius estimates for the sprinkler radius) of the zone
- mm/h value: average mm/h value from all sprinklers in the zone, multiplied by the m2 of the zone and divided by 60 to get the l/m value. (1 mm/m2 = 1l)

When I compare these values, the l/m value from the specsheet is significantly (+-100%) more than the mm/h value (adjusted to l/m) for zones with "rate-matched" nozzles (ie. nozzles that have constant precipitation rate) (see example specsheet attached)
[r-van-nozzles-performance-charts.pdf](https://github.com/user-attachments/files/17138252/r-van-nozzles-performance-charts.pdf)

I'm leaning more towards using the mm/h value or measure the zone mm/h rate using a rain gauge at multiple points in the zone.

Does anybody have advice on this?

**Comment by jeewee00** [comment]
@jeroenterheerdt @lynxchat4 & @lhsans , thanks for your replies.
Unfortunately I have two reasons why the suggested options don't seem to work for me:
1. The throughput depends on the sprinkler combined with the water pressure, therefore the documentation also gives a throughput bandwidth. E.g. [here](https://www.rainbird.com/products/3500-series-rotor-sprinklers) you see that the throughput is 2-17,4 liters per minute. Am I overseeing something?
2. Also I'm using self-pumped groundwater from a pump without a water meter, so I cannot let it run for X minutes and see how much water it used.

Maybe I oversee something, so I'd love to hear where my reasoning flaws.

**Comment by justusk18s** [comment]
I did it like this:
Take a picture of my water meter, which provides me with four decimals of m3 so 1234,5678m3
Then I let the sprinklers run for an hour and again take a picture of my water meter.
Now I can calculate the difference, convert to liters, and divide by 60, so it is liters per minute.

Please Note: Do this, when your regular irrigation time would be, as the waterpressure from the utility and therefore your throughput may vary. I usually have something like 2 bar of pressure during the day, which goes up to 3-4 bar at night, so this is when I irrigate.

Hope this helps

**Comment by Thorsten1982** [comment]
Hello, I have a Rainbird drip line and it has a flow rate of 2.3 l/h. Do I need to enter this value, or do I have to add all the drippers in the zone together and enter the total?

**Reply by justusk18s** [reply]
Exactly: you enter the total of all the drip line flow rates in one zone.
